tests/test_boundarylayer.py

```diff
@@ -0,0 +1,145 @@
+       -2.19, -4.74, -5.87, 0.32, -3.34, -0.71, 4.32, -0.97, 3.56, 5.26,
+       2.46, -1.88, -4.58, -3.45, 2.76, -3.63, 1.13, 4.07, -2.67, -4.55,
+       2.43, -0.38, -1.97, -0.58, -2.32, -1.24, 2.76, -1.25, -2.11, 3.64,
+       -3.37, -3.37, 3.03, 4.87, -0.88, 0.25, 0.52, -6.89, -3.69, -0.01,
+       1.7, 0.47, 0.46, 0.03, 0.17, 1.4, 1.97, -1.81, 0.1, -5.69, -5.77,
+       -6.36, -6.09, -0.36, -1.96, -5.48, 1.39, -0.93, -3.39, -0.04, 1.71, 
+       1.68, 1.59, -4.38, -2.79, -4.65, -1.84, -0.01, 3.58, -3.44, -4.42,
+       -8.27, -11.55, -10.06, -9.23, -9.33, -4.06, -5.7, -5.42, -5.49,
+       -6.71, -7.8, -4.16, -3.03, -8.58, -7.86, -4.35, -4.4, -4.45, -1.89
+    ]
+) * units.meter_per_second
+
+SAMPLE_RELATIVE_HUMIDITY = np.array(
+    [
+        48.5, 45.2, 45. , 46. , 47.6, 48.2, 49.4, 50.5, 51.6, 52.4, 53.2,
+        54.4, 55.3, 55.5, 56.7, 58.2, 59. , 60.6, 61.2, 62.9, 64.4, 65.7,
+        66.7, 67.1, 66.6, 66.9, 67. , 67.7, 69.6, 71.6, 72.8, 73.3, 75.5,
+        76.9, 78.1, 79.4, 81.3, 82.1, 80.6, 80.8, 82.9, 76.9, 62.1, 57.6,
+        29.1, 24.4, 21.9, 28.3, 28.5, 28.2, 28.1, 27.1, 28.7, 28.7, 25.8,
+        31.3, 31.6, 33.9, 32.4, 37. , 40. , 40.9, 37.7, 37.2, 41. , 43.4,
+        44.8, 49. , 50.6, 52.8, 49.9, 46.4, 39.4, 28.3, 30.7, 26.7, 23.8,
+        41.3, 45. , 48.2, 27.7, 18. , 14.8, 16.1, 17.3, 16.4, 16.2, 15.7,
+        14.9, 14.8, 14.5, 13.4, 12.4, 14.6, 15.3, 15.5, 15.3, 14.7, 13.8,
+        11.7
+    ]
+) * units.percent
+
+
+# Deduced data
+# ------------
+potential_temperature = mpcalc.potential_temperature(SAMPLE_PRESSURE, SAMPLE_TEMPERATURE)
+dewpoint = mpcalc.dewpoint_from_relative_humidity(SAMPLE_TEMPERATURE, SAMPLE_RELATIVE_HUMIDITY)
+specific_humidity = mpcalc.specific_humidity_from_dewpoint(SAMPLE_PRESSURE, dewpoint)
+
+
+# BOUNDARY LAYER HEIGHT ESTIMATIONS
+# =================================
+
+blh = boundarylayer.blh_from_richardson_bulk(SAMPLE_HEIGHT, potential_temperature, SAMPLE_U, SAMPLE_V)
+print(f"Estimated boundary layer height with blh_from_richardson_bulk:\t {blh}")
+
+blh = boundarylayer.blh_from_parcel(SAMPLE_HEIGHT, potential_temperature)
+print(f"Estimated boundary layer height with blh_from_parcel:\t {blh}")
+
+blh = boundarylayer.blh_from_humidity_gradient(SAMPLE_HEIGHT, SAMPLE_RELATIVE_HUMIDITY)
+print(f"Estimated boundary layer height with blh_from_humidity_gradient (relative humidity):\t {blh}")
+
+blh = boundarylayer.blh_from_humidity_gradient(SAMPLE_HEIGHT, specific_humidity)
+print(f"Estimated boundary layer height with blh_from_humidity_gradient (specific humidity):\t {blh}")
+
+blh = boundarylayer.blh_from_temperature_inversion(SAMPLE_HEIGHT, SAMPLE_TEMPERATURE)
+print(f"Estimated boundary layer height with blh_from_temperature_inversion (absolute temperature):\t {blh}")
+
+blh = boundarylayer.blh_from_temperature_inversion(SAMPLE_HEIGHT, potential_temperature)
+print(f"Estimated boundary layer height with blh_from_temperature_inversion (potential temperature):\t {blh}")
```

Comment:
Sure, I can write the corresponding `pytest` unitary tests based on these computations. However, I better use a standard Metpy profile rather than my own testing data. Do you know where I could find this sounding data?

Reply:
Thank you very much for the review, @DWesl !

Reply:
https://github.com/Unidata/MetPy/tree/main/staticdata has a few files matching `*_sounding.txt` that look promising. Those are used here:
https://github.com/Unidata/MetPy/blob/eaf608d6c65bc7f6f30d8ac440de1f5cea9fd842/examples/Advanced_Sounding.py#L30-L31

The [SkewT tests](https://github.com/Unidata/MetPy/blob/eaf608d6c65bc7f6f30d8ac440de1f5cea9fd842/tests/plots/test_skewt.py#L27-L29) look like they use `p = np.geomspace(1000, 100, 10); T = np.linspace(20, -20, 10)` and similar profiles.  You could build something similar for a first test (constant potential temperature and water vapor mixing ratio up to some height, then a sharp increase in the former and a sharp decrease in the latter)